login/pkg/handlers/forgot_password.go

```diff
@@ -0,0 +1,56 @@
+package handlers
+
+import (
+	"net/http"
+
+	"github.com/verasthiago/tickets-generator/shared/auth"
+	"github.com/verasthiago/tickets-generator/shared/models"
+
+	"github.com/gin-gonic/gin"
+	"github.com/verasthiago/tickets-generator/login/pkg/builder"
+	"github.com/verasthiago/tickets-generator/login/pkg/validator"
+	error_handler "github.com/verasthiago/tickets-generator/shared/errors"
+)
+
+type ForgotPasswordAPI interface {
+	Handler(context *gin.Context)
+}
+
+type ForgotPasswordHandler struct {
+	builder.Builder
+}
+
+func (f *ForgotPasswordHandler) InitFromBuilder(builder builder.Builder) *ForgotPasswordHandler {
+	f.Builder = builder
+	return f
+}
+
+func (f *ForgotPasswordHandler) Handler(context *gin.Context) {
+	var request *validator.ForgotPasswordRequest
+	var user *models.User
+	var err error
+	var tokenString string
+
+	if err := context.ShouldBindJSON(&request); err != nil {
+		error_handler.HandleBadRequestError(context, err)
+		return
+	}
+
+	if errors := request.Validate(); len(errors) > 0 {
+		error_handler.HandleBadRequestErrors(context, errors)
+		return
+	}
+
+	if user, err = f.GetRepository().GetUserByEmail(request.Email); err != nil {
+		error_handler.HandleInternalServerError(context, err, f.GetLog())
+		return
+	}
+
+	if tokenString, err = auth.GenerateJWT(user.Email, user.Username, user.ID, f.GetSharedFlags().JwtKey, user.IsAdmin); err != nil {
```

Comment:
You are using the normal jwt key

Reply:
i don't know what happened but i think u were looking to old file version